Deploy more smart pointers in UserMediaCaptureManager.cpp

abigailfox opened the pull request:
#### 0f291f6f041e215173fe2e19bf20648b10a73c36
<pre>
Deploy more smart pointers in UserMediaCaptureManager.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=266030">https://bugs.webkit.org/show_bug.cgi?id=266030</a>
<a href="https://rdar.apple.com/119343045">rdar://119343045</a>

Reviewed by Chris Dumez.

* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::DisplayFactory::createDisplayCaptureSource):

Canonical link: <a href="https://commits.webkit.org/271755@main">https://commits.webkit.org/271755@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0b555233625dcdbbfc2957ce508b5b96927ec897

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29558 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8224 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30879 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32084 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26772 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10376 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5517 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29831 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/6866 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/25245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5868 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33428 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/26969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/26764 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/32220 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/5956 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/4148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/29995 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/7692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7012 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/6498 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/6479 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
```diff
@@ -233,8 +233,10 @@ CaptureSourceOrError UserMediaCaptureManager::DisplayFactory::createDisplayCaptu
-        m_manager.m_remoteCaptureSampleManager.setVideoFrameObjectHeapProxy(&WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy());
+    if (m_shouldCaptureInGPUProcess) {
+        RefPtr videoFrameObjectHeapProxy = &WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy();
+        m_manager.m_remoteCaptureSampleManager.setVideoFrameObjectHeapProxy(videoFrameObjectHeapProxy.get());
```
Please pass `WTFMove(videoFrameObjectHeapProxy)` here and update `setVideoFrameObjectHeapProxy()` to take in a `RefPtr<RemoteVideoFrameObjectHeapProxy>&&` and `WTFMove()` inside it.

This will avoid refcounting churn.

cdumez commented on Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
```diff
@@ -233,8 +233,10 @@ CaptureSourceOrError UserMediaCaptureManager::DisplayFactory::createDisplayCaptu
-    if (m_shouldCaptureInGPUProcess)
-        m_manager.m_remoteCaptureSampleManager.setVideoFrameObjectHeapProxy(&WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy());
+    if (m_shouldCaptureInGPUProcess) {
+        RefPtr videoFrameObjectHeapProxy = &WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy();
```
`Ref videoFrameObjectHeapProxy = WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy();`

cdumez reviewed (approved):
LGTM

abigailfox committed:
Deploy more smart pointers in UserMediaCaptureManager.cpp
https://bugs.webkit.org/show_bug.cgi?id=266030
rdar://119343045

Reviewed by Chris Dumez.

* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::DisplayFactory::createDisplayCaptureSource):

Canonical link: https://commits.webkit.org/271755@main
